usaspending_api/spending_explorer/v2/filters/spending_filter.py

```diff
@@ -0,0 +1,331 @@
+import logging
+import urllib
+from urllib.error import HTTPError
+
+from usaspending_api.awards.models import Award
+from usaspending_api.common.exceptions import InvalidParameterException
+
+logger = logging.getLogger(__name__)
+
+
+def spending_filter(alt_set, queryset, filters, _type):
+
+    for key, value in filters.items():
+        # check for valid key
+        if value is None:
+            raise InvalidParameterException('Invalid filter: ' + key + ' has null as its value.')
+
+        key_list = ['budget_function', 'budget_subfunction', 'federal_account', 'program_activity', 'object_class',
+                    'recipient', 'award', 'award_category', 'agency', 'agency_type', 'agency_sub', 'fy']
+
+        if key not in key_list:
+            raise InvalidParameterException(key + ' filter does not exist. '
+                                                  'Valid Filters: budget_function, budget_subfunction, federal_account,'
+                                                  'program_activity, object_class, recipient, award, award_category,'
+                                                  'agency, agency_type, agency_sub fy.')
+
+        # Check _type to filter on correct set (alt_set or queryset)
+        alt_set_keys = ['recipient', 'award', 'award_category', 'agency_type', 'agency_sub']
+        if _type in alt_set_keys:
+            # Apply filters
+            # budget_function
+            if key == 'budget_function':
+                or_alt_set = None
+                if or_alt_set:
+                    or_alt_set |= or_alt_set.filter(
```

Comment:
For the `or_alt_set.filter`s and `alt_set.filter`s, can we replace those with `FinancialAccountsByAwards.objects.filter`s? Otherwise that caused some errors with our award and transaction filters for the award searches.